src/Common/ProcessExecutor.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Robo\Common;
+
+use Robo\Contract\OutputAdapterInterface;
+use Robo\Contract\OutputAwareInterface;
+use Robo\Contract\TaskInterface;
+use Robo\Contract\VerbosityThresholdInterface;
+use Robo\Robo;
+use Robo\Task\BaseTask;
+
+class ProcessExecutor implements OutputAwareInterface
+{
+    use ExecTrait;
+    use TaskIO; // uses LoggerAwareTrait and ConfigAwareTrait
+    use ProgressIndicatorAwareTrait;
+    use OutputAwareTrait;
+
+    public function __construct($process)
+    {
+        $this->process = $process;
+        $this->logger = Robo::logger();
+        $this->setOutputAdapter(new OutputAdapter());
+        $this->outputAdapter()->setOutput(Robo::output());
```

Comment:
We are using two static calls in this constructor. This was a very efficient and direct way to achieve the desired behavior. Let me know if there's a preferred alternative.